fw/sock.c

```diff
@@ -421,16 +428,21 @@ ss_skb_tcp_entail(struct sock *sk, struct sk_buff *skb, unsigned int mark,
 }
 
 int
-ss_skb_tcp_entail_list(struct sock *sk, struct sk_buff **skb_head)
+ss_skb_tcp_entail_list(struct sock *sk, struct sk_buff **skb_head,
+		       unsigned int mss_now, unsigned long *snd_wnd)
 {
-	struct sk_buff *skb, *tail, *next, *to_destroy;
+	struct sk_buff *tail, *next, *to_destroy;
 	unsigned char tls_type = 0;
 	unsigned int mark = 0;
 	void *opaque_data = NULL;
 	void (*destructor)(void *) = NULL;
 	int r;
 
-	while ((skb = ss_skb_dequeue(skb_head))) {
+	while ((*snd_wnd = tfw_tcp_calc_snd_wnd(sk, mss_now))) {
```

Comment:
We calculate window here, but we don't check it in the loop. I mean we can put into a `write_queue` skb with len 200 when `snd_wnd` is 1.